src/mame/drivers/bublbobl.cpp

```diff
@@ -942,16 +982,34 @@ static MACHINE_CONFIG_START( bublbobl )
 	/* sound hardware */
 	MCFG_SPEAKER_STANDARD_MONO("mono")
 
-	MCFG_GENERIC_LATCH_8_ADD("soundlatch")
-
 	MCFG_SOUND_ADD("ym1", YM2203, MAIN_XTAL/8)
-	MCFG_YM2203_IRQ_HANDLER(INPUTLINE("audiocpu", 0))
+	//MCFG_YM2203_IRQ_HANDLER(INPUTLINE("audiocpu", 0))
+	MCFG_YM2203_IRQ_HANDLER(WRITELINE(bublbobl_state, ym2203_irqhandler))
 	MCFG_SOUND_ROUTE(ALL_OUTPUTS, "mono", 0.25)
 
 	MCFG_SOUND_ADD("ym2", YM3526, MAIN_XTAL/8)
+	MCFG_YM3526_IRQ_HANDLER(WRITELINE(bublbobl_state, ym3526_irqhandler))
 	MCFG_SOUND_ROUTE(ALL_OUTPUTS, "mono", 0.50)
 MACHINE_CONFIG_END
 
+WRITE_LINE_MEMBER(bublbobl_state::ym2203_irqhandler)
```

Comment:
You should be able to use an `input_merger_device` in place of this glue and define it all in machine config.  The members would be unnecessary.